lib/Backends/CPU/CPUFunction.cpp

```diff
@@ -13,28 +13,87 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 #include "CPUFunction.h"
 
+#include "glow/Graph/Context.h"
 #include "glow/Support/Compiler.h"
 #include "glow/Support/Memory.h"
 
 using namespace glow;
 
-CPUFunction::CPUFunction(std::unique_ptr<llvm::orc::GlowJIT> JIT, void *heap)
-    : JIT_(std::move(JIT)), heap_(heap) {}
+CPUFunction::CPUFunction(std::unique_ptr<llvm::orc::GlowJIT> JIT,
+                         std::unique_ptr<runtimeInfo> runtimeInfo)
+    : JIT_(std::move(JIT)), runtimeInfo_(std::move(runtimeInfo)) {}
+
+CPUFunction::~CPUFunction() { free(runtimeInfo_->constants); }
+
+void CPUFunction::loadFunctionOnDevice() {
+  baseConstantWeightVarsAddress_ = (uint8_t *)alignedAlloc(
+      runtimeInfo_->constantWeightVarsMemSize, TensorAlignment);
+  // Copy Constants to device
+  memcpy(baseConstantWeightVarsAddress_, runtimeInfo_->constants,
+         runtimeInfo_->constantWeightVarsMemSize);
+}
+
+void CPUFunction::allocateMutableBuffersOnDevice() {
+  baseActivationsAddress_ = (uint8_t *)alignedAlloc(
+      runtimeInfo_->activationsMemSize, TensorAlignment);
+
+  baseMutableWeightVarsAddress_ = (uint8_t *)alignedAlloc(
+      runtimeInfo_->mutableWeightVarsMemSize, TensorAlignment);
+}
+
+void CPUFunction::loadMutablesOnDevice(Context &ctx) {
+  // Copy Placeholders into allocated memory.
+  for (auto PH : ctx.pairs()) {
+    auto payload = PH.second->getUnsafePtr();
+    auto symbolInfo =
+        runtimeInfo_->symbolTable.find(std::string(PH.first->getName()));
+    assert(symbolInfo != runtimeInfo_->symbolTable.end() && "Symbol not found");
+    auto addr = symbolInfo->second.offset;
+    auto numBytes = symbolInfo->second.size;
+    // copy PH to allocated memory.
+    memcpy(baseMutableWeightVarsAddress_ + addr, payload, numBytes);
+  }
+}
+
+void CPUFunction::fetchFromDevice(Context &ctx) {
```

Comment:
Since you have a `loadMutablesOnDevice` method, the name for this one should be more symmetric, e.g. `fetchMutablesFromDevice`.

But in other backends, we usualy use names like 'copySomethingToDevice` and `copySomethingFromDevice`.